packages/frontend/src/state/controller/hooks.ts

```diff
@@ -210,6 +210,17 @@ export const useGetTwapEthPrice = () => {
   return getTwapEthPrice
 }
 
+export const useGetTwapSqueethPrice = () => {
+  const { squeethPool, oSqueeth, weth } = useAtomValue(addressesAtom)
+  const { getTwapSafe } = useOracle()
+  const getTwapSqueethPrice = useCallback(
+    async () => await getTwapSafe(squeethPool, oSqueeth, weth, TWAP_PERIOD),
+    [oSqueeth, weth, getTwapSafe, squeethPool],
+  )
+
+  return getTwapSqueethPrice
```

Comment:
Not a big deal but can we use useMemo instead and return twapPrice directly as sqeeuthPool, oSqueeth, weth won't change?
const twapSqueethPrice = useMemo(() => ...)
Just wanted to avoid performance issues due to calling getTwapSqueethPrice frequently

Reply:
Sorry, useMemo can't be used here because getTwapSafe will return a different value depending on when it's called. There's a different solution for avoiding frequent API calls using useQuery.
https://github.com/opynfinance/squeeth-monorepo/pull/257/commits/c51e11460c9e5908a5feffc76c709b0d8d4971ec
But it shouldn't be used in LP-related hooks. So everything is good with the current implementation.

Reply:
Sounds good, will keep the current implementation. Why can't we use useQuery in LP-related hooks?

Reply:
It uses cached data, and the cache will be updated every 10 seconds, so it's not perfectly accurate.

This hook is good enough for resolving this issue https://github.com/opynfinance/squeeth-monorepo/pull/257#issuecomment-1177025880 Alexis mentioned. The issue was related to fetching the twap price too frequently when the slider value changed, and this hook allows us to use previously fetched data assuming there won't be much change.

But I think LP required perfect accuracy and LP won't make frequent twap price calls so it doesn't seem to make sense to use stale data.